libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -60,45 +64,23 @@ using __type_call2 _LIBCUDACXX_NODEBUG_TYPE = typename _Fn::template __call<_Ty,
 
 namespace __detail
 {
-// \brief A helper for determining the arity of a template.
-// (This is unused for now, I just want to see if this passes CI.)
-template <template <class> class _Fn>
-_CCCL_HOST_DEVICE constexpr int __template_arity_aux()
-{
-  return 1;
-}
-
-template <template <class, class> class _Fn>
-_CCCL_HOST_DEVICE constexpr int __template_arity_aux()
+template <size_t _DependentValue>
+struct __type_call_indirect_fn
 {
-  return 2;
-}
-
-template <template <class...> class _Fn, int _Arity = __template_arity_aux<_Fn>()>
-_CCCL_HOST_DEVICE constexpr int __template_arity(int)
-{
-  return _Arity;
-}
-
-template <template <class...> class _Fn>
-_CCCL_HOST_DEVICE constexpr int __template_arity(long)
-{
-  return -1;
-}
-
-template <class...>
-struct _Tn;
-template <class>
-struct _T1;
-template <class, class>
-struct _T2;
-static_assert(__template_arity<_Tn>(0) == -1, "");
-static_assert(__template_arity<_T1>(0) == 1, "");
-static_assert(__template_arity<_T2>(0) == 2, "");
+  template <template <class...> class _Fn, class... _Ts>
+  using __call _LIBCUDACXX_NODEBUG_TYPE = _Fn<_Ts...>;
+};
 } // namespace __detail
 
+//! \brief Evaluate a meta-callable with the given arguments, with an indirection
+//! to avoid the dreaded "pack expansion argument for non-pack parameter" error.
```

Comment:
Life if we could pass single types to packs
![2ynjel](https://github.com/user-attachments/assets/b31889ff-dc70-4960-b529-4c16c11bc965)